src/Workspaces/MSBuildTest/MSBuildWorkspaceTests.cs

```diff
@@ -254,6 +254,21 @@ public async Task TestOutputFilePaths()
             Assert.Equal("VisualBasicProject.dll", Path.GetFileName(p2.OutputFilePath));
         }
 
+        [ConditionalFact(typeof(VisualStudioMSBuildInstalled)), Trait(Traits.Feature, Traits.Features.MSBuildWorkspace)]
+        public async Task TestCompilationOutputInfo()
+        {
+            CreateFiles(GetMultiProjectSolutionFiles());
+            var solutionFilePath = GetSolutionFileName("TestSolution.sln");
+
+            using var workspace = CreateMSBuildWorkspace();
+            var sol = await workspace.OpenSolutionAsync(solutionFilePath);
+            var p1 = sol.Projects.First(p => p.Language == LanguageNames.CSharp);
+            var p2 = sol.Projects.First(p => p.Language == LanguageNames.VisualBasic);
+
+            Assert.Equal("CSharpProject.dll", Path.GetFileName(p1.CompilationOutputInfo.AssemblyPath));
```

Comment:
Can we actually validate the full path here?

Reply:
I copied the same assertions as the one from TestOutputFilePaths: https://github.com/dotnet/roslyn/pull/51671/files#diff-580f2cc883f4329f6301c2f8e2285e1cb4ec18a8072e4439311135ca51f93bf1L253-L254. I couldn't tell if was intentionally avoiding the full path comparison (OS oddities)?